src/dkg_commitmentmatrix.erl

```diff
@@ -79,3 +79,16 @@ public_key_share(U, Matrix, NodeID) ->
                                           end, G1, lists:reverse(lists:seq(1, tuple_size(Matrix)))),
                         erlang_pbc:element_mul(R, Row)
                 end, G1, lists:reverse(lists:seq(1, tuple_size(Matrix)))).
+
+serialize(Matrix) ->
+    lists:foldl(fun({I, J}, Acc) ->
+                      insert([I, J], Acc, erlang_pbc:element_to_binary(lookup([I,J], Acc)))
+              end, Matrix,
+      [ {I, J} || I <- lists:seq(1, tuple_size(Matrix)), J <- lists:seq(1, tuple_size(Matrix))]).
```

Comment:
I'd suggest having a `size` and `iter` function in this module as well. Would reduce code duplication a bit.